classes/items/weapon.hxx

```diff
@@ -17,9 +17,9 @@ private:
     public:
     Weapon(std::string name ="Weapon",
             int damage = 0,
-            unsigned int type=1,
-            unsigned int tier=0,
+            unsigned int type = 1,
             unsigned int value=0,
+            unsigned int tier=0,
             unsigned int level=1);
```

Comment:
Same as the item class. Change type to be not-defined here. Also default params should be in cpp file.